### PR TITLE
DG-15622, Extended site groups API

### PR DESCRIPTION
**Pull request by dmitrybezer**

### CODE REVIEW

File: lib/sfrest/connection.rb [hunk 1]
```diff
@@ -80,13 +80,14 @@ def put(uri, payload)
     # @return [Object] ruby representation of the json response
     #                  if the reponse body  does not parse, raises
     #                  a SFRest::InvalidResponse
-    def delete(uri)
+    def delete(uri, payload = '')
```
Comment (nikgregory) [inline lib/sfrest/connection.rb]: Add a description regarding payload.

File: lib/sfrest/group.rb [hunk 2]
```diff
@@ -15,6 +15,13 @@ def create_group(groupname)
       @conn.post(current_path, payload)
     end
 
+    # Deletes the group with the specified id
+    # @param [Integer] group_id Id of the group to fetch
+    def delete_group(group_id)
```
Comment (nikgregory) [inline lib/sfrest/group.rb]: What is the default behavior if the group is deleted? I know it is defined by the php code but isn't there some questions what  to do with the sites/collections that are in the group?

File: lib/sfrest/connection.rb (hunk 1 above)
Comment (nikgregory) [inline lib/sfrest/connection.rb]: Add a spec for the new param

Comment (dmitrybezer) [reply lib/sfrest/connection.rb]: Comment added.

Comment (dmitrybezer) [reply lib/sfrest/connection.rb]: Added.

File: lib/sfrest/group.rb (hunk 2 above)
Comment (dmitrybezer) [reply lib/sfrest/group.rb]: As far as I can tell, the member sites and users are excluded from the group when it is deleted.

**Pull request merged**